Add NeoLoad integration (#1587)
DataDog/integrations-extras: neoload/README.md

# NeoLoad integration

## Overview

[Tricentis NeoLoad][1] simplifies and scales performance testing for APIs and microservices, as well as end-to-end application testing through protocol and browser-based capabilities.

With the NeoLoad integration, you can track performance metrics of NeoLoad tests to:

- Correlate application performance with load testing metrics.

> - Analyze and visualize NeoLoad metrics using the out-of-the-box Dashboard or [Metrics Explorer][7].

Reviewer: ```suggestion
- Analyze and visualize NeoLoad metrics using the out-of-the-box dashboard or [Metrics Explorer][7].
```

> [Tricentis NeoLoad][1] simplifies and scales performance testing for APIs and microservices, as well as end-to-end application testing through protocol and browser-based capabilities.
> 
> With the NeoLoad integration, you can track performance metrics of NeoLoad tests to:
> 
> - Correlate application performance with load testing metrics.
> - Analyze and visualize NeoLoad metrics using the out-of-the-box Dashboard or [Metrics Explorer][7].

Reviewer 2: Added additional information on what this integration tracks based off the demo today. Please edit if I misunderstood

```suggestion
[Tricentis NeoLoad][1] simplifies and scales performance testing for APIs and microservices, as well as end-to-end application testing through protocol and browser-based capabilities.

With the NeoLoad integration, you can track Neoload test performance metrics to:

- Correlate application performance with load testing metrics in Neoload.
- Analyze and visualize NeoLoad metrics in Datadog like throughput, errors, and performance using the out-of-the-box Dashboard or [Metrics Explorer][7].
```

## Setup

### Configuration

> For detailed instructions on NeoLoad configuration, follow the [NeoLoad documentation][2]. Choose which metrics to send in the "Push Counters" configuration of the Datadog Connector in NeoLoad. Available since NeoLoad 9.1.

Reviewer: ```suggestion
For detailed instructions on NeoLoad configuration, follow the [NeoLoad documentation][2]. Since NeoLoad version 9.1, you can choose which metrics to send in the **Push Counters** configuration of the Datadog Connector in NeoLoad.
```

When Datadog receives `NeoLoad.*` metrics, the NeoLoad integration tile is installed automatically, and the default NeoLoad dashboard is added to your dashboard list.

## Data Collected

### Metrics

See [metadata.csv][3] for a list of metrics provided by this integration.

### Events

All NeoLoad performance tests events are sent to your [Datadog Events Explorer][4].
NeoLoad sends events to the Datadog API when a performance test starts and ends.

> Set the option in the "Push Counters" configuration of the Datadog Connector in NeoLoad. Available since NeoLoad 9.1.

Reviewer 2: Is this setting for sending events to Datadog? Can you clarify what this setting does?

Author: Yes this setting enable/disable sending events to Datadog. It is enabled by default, and it is pretty obvious for the user lookig at NeoLoad UI.

## Troubleshooting

Need help? Contact [Datadog support][5] or [Tricentis NeoLoad support][6].

[1]: https://www.tricentis.com/products/performance-testing-neoload
[2]: https://documentation.tricentis.com/neoload/9.1/en/content/reference_guide/datadog.htm
[3]: https://github.com/DataDog/integrations-extras/blob/master/neoload/metadata.csv
[4]: https://docs.datadoghq.com/events/
[5]: https://docs.datadoghq.com/help/
[6]: https://support-hub.tricentis.com/

> [7]: https://app.datadoghq.com/metric/explorer

Reviewer 3: ```suggestion
[7]: /metric/explorer
```

# NeoLoad integration

## Overview

[Tricentis NeoLoad][1] simplifies and scales performance testing for APIs and microservices, as well as end-to-end application testing through protocol and browser-based capabilities.

With the NeoLoad integration, you can track performance metrics of NeoLoad tests to:

- Correlate application performance with load testing metrics.
- Analyze and visualize NeoLoad metrics using the NeoLoad Datadog Dashboard or Metrics Explorer.

## Setup

### Configuration

For detailed instructions on NeoLoad configuration, follow the [NeoLoad documentation][2].
When Datadog receives `NeoLoad.*` metrics, the NeoLoad integration tile is installed automatically, and the default NeoLoad dashboard is added to your dashboard list.

![NeoLoad Dashboard][7]

## Data Collected

### Metrics

See [metadata.csv][3] for a list of metrics provided by this integration.

### Service Checks

NeoLoad does not include any service checks.

### Events

All NeoLoad performance tests events are sent to your [Datadog Events Explorer][4].

> NeoLoad sends events to the Datadog API when a performance test starts and ends.

Author: Why remove this sentence that tells the user what kinds of events are sent and when ?

# NeoLoad integration

## Overview

> [Tricentis NeoLoad][1] simplifies and scales performance testing for everything, from APIs and microservices, to end-to-end application testing through innovative protocol and browser-based capabilities.

Reviewer: ```suggestion
[Tricentis NeoLoad][1] simplifies and scales performance testing for APIs and microservices, as well as end-to-end application testing through protocol and browser-based capabilities.
```

With the NeoLoad integration, you can track performance metrics of NeoLoad tests to:

- Correlate application performance with load testing metrics.
- Analyze and visualize NeoLoad metrics using the NeoLoad Datadog Dashboard or Metrics Explorer.

## Setup

### Configuration

> The Datadog agent is not needed to use the integration.
> For the detailed instructions on NeoLoad configuration, follow the [NeoLoad documentation][2].
> Additionally, the first time Datadog detects the NeoLoad.Controller.User.Load metric, the NeoLoad integration tile is installed automatically, and the default NeoLoad dashboard is added to your dashboard list.

Reviewer: ```suggestion
For detailed instructions on NeoLoad configuration, follow the [NeoLoad documentation][2].
When Datadog receives `NeoLoad.*` metrics, the NeoLoad integration tile is installed automatically, and the default NeoLoad dashboard is added to your dashboard list.
```

Reviewer: Could you provide more detail about how the user sends `Neoload.*` metrics to Datadog? The provided link leads to multiple options, so any additional clarification here will be helpful for our users.

Author: You're right, there is a few missing informations, I will add this.
To send events and to send metrics with NeoLoad, the user must use the version 9.1 of NeoLoad that should be published shortly (it is ready waiting for approvals).
The link to the doc will be updated when publishing with more details on the integration.
It will contain a screenshot (the user can choose what to send. Everything is checked by default).

Author: ![image](https://user-images.githubusercontent.com/57721694/199698551-cd7a62c3-61cc-4b91-b1ee-96f5c482317e.png)

![NeoLoad Dashboard][7]

## Data Collected

### Metrics

See [metadata.csv][3] for a list of metrics provided by this integration.

### Service Checks

NeoLoad does not include any service checks.

### Events

> All NeoLoad performance tests events are sent to your [Datadog Event Stream][4].

Reviewer: ```suggestion
All NeoLoad performance tests events are sent to your [Datadog Events Explorer][4].
```

> NeoLoad send events to Datadog via the API when a performance test starts and ends.

Reviewer: ```suggestion
NeoLoad sends events to the Datadog API when a performance test starts and ends.
```

# NeoLoad integration

## Overview

[Tricentis NeoLoad][1] simplifies and scales performance testing for APIs and microservices, as well as end-to-end application testing through protocol and browser-based capabilities.

With the NeoLoad integration, you can track performance metrics of NeoLoad tests to:

- Correlate application performance with load testing metrics.
- Analyze and visualize NeoLoad metrics using the NeoLoad Datadog Dashboard or Metrics Explorer.

## Setup

### Configuration

For detailed instructions on NeoLoad configuration, follow the [NeoLoad documentation][2]. Choose which metrics to send in the "Push Counters" configuration of the Datadog Connector in NeoLoad. Available since NeoLoad 9.1.

> When Datadog receives `NeoLoad.*` metrics, the NeoLoad integration tile is installed automatically, and the default NeoLoad dashboard is added to your dashboard list.

Reviewer 2: Users will need to click the install button in Datadog in order to see the dashboard.

Author: Ok, what configuration is needed to install automatically the integration ?
We used docker integration, and MySQL and the Dashboards were added automcatically in the dashboard's list. It would be nice for us if the NeoLoad dashboard is added when Datadog receives metrics from NeoLoad. Do you know how to d othat ?

Reviewer 2: Hey @guillaumebert - Customers will need to:
1. Go to the [integration](https://app.datadoghq.com/integrations) page in datadog
2. Click on the NeoLoad tile (once it is pushed live) 
3. Click install 

Once they do that, the dashboard will appear. There isn't a way to accomplish this without having customers click the "install" button. For further context, the install button adds the customers account and uninstall button will remove the account from the integration.

Reviewer 2: @guillaumebert it looks like this still needs to be updated. Could you make the changes we spoke about above?

![NeoLoad Dashboard][7]

## Data Collected

### Metrics

See [metadata.csv][3] for a list of metrics provided by this integration.

### Service Checks

NeoLoad does not include any service checks.

### Events

All NeoLoad performance tests events are sent to your [Datadog Events Explorer][4].
NeoLoad sends events to the Datadog API when a performance test starts and ends.
Set the option in the "Push Counters" configuration of the Datadog Connector in NeoLoad. Available since NeoLoad 9.1.

## Troubleshooting

Need help? Contact [Datadog support][5] or [Tricentis NeoLoad support][6].

[1]: https://www.tricentis.com/products/performance-testing-neoload

> [2]: https://documentation.tricentis.com/neoload/9.1/en/content/reference_guide/datadog.htm

Reviewer 2: This link redirects to another [link](https://documentation.tricentis.com/neoload/9.1/en/content/reference_guide/datadog.htm )

Can you update the existing link with the one above?
